src/main/java/com/dev/bankingservice/repository/RoleRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.dev.bankingservice.repository;
+
+import com.dev.bankingservice.entity.Role;
+import javax.persistence.Id;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface RoleRepository extends JpaRepository<Role, Id> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = :name")
+    Role getByName(@Param("name") Role.RoleName name);
```

Comment:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

Reply:
Thank you, done